src/dra/4A538.c

```diff
@@ -719,84 +719,81 @@ void ResetEntityArray(void) {
     }
 }
 
-#ifndef NON_MATCHING
-INCLUDE_ASM("dra/nonmatchings/4A538", RenderEntities);
-#else
 typedef struct {
-    /* 0x1F800034 */ u16 unk0;
-    /* 0x1F800036 */ s16 animSet;
-    /* 0x1F800038 */ s16 animCurFrame;
-    /* 0x1F80003A */ u16 w;
-    /* 0x1F80003C */ u16 h;
-    /* 0x1F80003E */ s16 x;
-    /* 0x1F800040 */ s16 y;
-    /* 0x1F800042 */ u16 xPivot;
-    /* 0x1F800044 */ u16 yPivot;
-    /* 0x1F800046 */ u16 unused12;
-    /* 0x1F800048 */ s32 index;
-    /* 0x1F80004C */ s32 unk18;
-    /* 0x1F800050 */ u32 unk1C;
-    /* 0x1F800054 */ s32 unk20;
-    /* 0x1F800058 */ s32 unk24;
-    /* 0x1F80005C */ OT_TYPE* ot;
-    /* 0x1F800060 */ POLY_GT4* poly;
-} EntitiesRenderer; // size=0x30
+    s16 unk0;
+    s16 animSet;
+    s16 animCurFrame;
+    u16 w;
+    u16 h;
+    s16 x;
+    s16 y;
+    s16 xPivot;
+    s16 yPivot;
+    u16 unused12;
+    s32 index;
+    s32 index2;
+    u32 spriteSheetIdx;
+    s32 flipX; // set to 2 when facing left, 0 when not
+    s32 eDrawFlags;
+    OT_TYPE* ot;
+    POLY_GT4* poly;
+} EntitiesRenderer;
 
 void RenderEntities(void) {
+    s16 polyX;
+    s16 polyY;
+    s32 animFrameFlags;
+    Entity* entity;
+    s16* animFrame; // might be a struct
+    u8 uvRight;
+    u8 uvTop;
+    s32 tempFlagCheck;
+    s16** spriteBank;
+    u8* spriteData;
+    PlayerDraw* plDraw;
+    u16 palette;
+    u16 animFrameIdx;
+    u8 uvLeft;
+    u8 uvBottom;
 #ifdef VERSION_PC
     POLY_GT4 _poly;
     POLY_GT4* poly = &_poly;
     EntitiesRenderer _r;
     EntitiesRenderer* r = &_r;
 #else
-    POLY_GT4* poly = (POLY_GT4*)0x1F800000;
-    EntitiesRenderer* r = (EntitiesRenderer*)0x1F800034;
+    POLY_GT4* poly;
+    EntitiesRenderer* r;
+    poly = (POLY_GT4*)SPAD(0);
+    setPolyGT4(poly);
```

Comment:
This needs to go outside the `#ifdef / #else / #endif` block, otherwise there will be no entity rendering

Reply:
Unfortunately the setPolyGT4 line is sandwiched between `poly` and `r` lines, so I decided to just duplicate it in both blocks, rather than having two separate sets of #ifdef/#else. Hopefully that's a suitable solution.